### PR TITLE
fixed few package issues. update travis.yml

### PR DESCRIPTION
**Pull request by rox163**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
* fixed few package issues. 
* updated travis.yml.

### CODE REVIEW

File: tests/dummy/app/app.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import Ember from 'ember'
-import Resolver from 'ember/resolver'
+const {Application} = Ember
 import loadInitializers from 'ember/load-initializers'
+import Resolver from 'ember/resolver'
```
Comment (juwara0) [inline tests/dummy/app/app.js]: We need to update this line to: `import Resolver from './resolver'` it was missed in previous upgrades.

File: tests/dummy/app/app.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import Ember from 'ember'
-import Resolver from 'ember/resolver'
+const {Application} = Ember
 import loadInitializers from 'ember/load-initializers'
```
Comment (juwara0) [inline tests/dummy/app/app.js]: We need to update this line to: `import loadInitializers from 'ember-load-initializers'` it was missed in previous upgrades.

File: tests/dummy/app/router.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import Ember from 'ember'
+const {Router} = Ember
```
Comment (juwara0) [inline tests/dummy/app/router.js]: import config from './config/environment'
import Ember from 'ember'
const {Router: EmberRouter} = Ember

const Router = EmberRouter.extend({
  location: config.locationType
})

Router.map(function () {
  this.route('demo', {path: '/'})
})

export default Router

Comment (juwara0) [inline tests/dummy/app/router.js]: You can destructure `Ember.Router()` and create an alias `EmberRouter` like this.

File: package.json [hunk 4]
```diff
@@ -16,42 +16,42 @@
...
     "ember-ajax": "0.7.1",
     "ember-cli": "~2.8.0",
     "ember-cli-app-version": "^1.0.0",
+    "ember-cli-chai": "0.3.2",
     "ember-cli-dependency-checker": "^1.2.0",
+    "ember-cli-frost-blueprints": "1.2.1",
     "ember-cli-htmlbars-inline-precompile": "^0.3.1",
     "ember-cli-inject-live-reload": "^1.4.0",
-    "ember-cli-jshint": "^1.0.0",
+    "ember-cli-mocha": "^0.10.0",
```
Comment (juwara0) [inline package.json]: This can be set to: `"ember-cli-mocha": "~0.13.2",`

File: bower.json [hunk 5]
```diff
@@ -3,9 +3,9 @@
...
-    "Faker": "3.1.0",
     "perfect-scrollbar": "~0.6.10",
-    "pretender": "~0.12.0"
-  },
-  "resolutions": {}
-}
\ No newline at end of file
+    "mocha": "~2.2.4",
+    "chai": "~2.3.0",
+    "ember-mocha-adapter": "~0.3.1"
```
Comment (juwara0) [inline bower.json]: mocha, chai and ember-mocha-adapter can be removed once this `"ember-cli-mocha": "~0.13.2",` is set in the `package.json`

File: .travis.yml [hunk 6]
```diff
@@ -1,20 +1,45 @@
...
+
+cache:
+  directories:
+  - node_modules
+
 before_install:
-- npm install -g coveralls pr-bumper
-- pr-bumper check
+- npm config set spin false
+- npm install -g coveralls pr-bumper@^1.0.0
```
Comment (juwara0) [inline .travis.yml]: Lets go ahead and make this use the newer version 2:
`- npm install -g coveralls pr-bumper@^2.0.0`

We will need to create the `.pr-bumper.json` file as well. It should have this:
```
{
  "features": {
    "changelog": {
      "enabled": true
    }
  }
}

```

File: .travis.yml [hunk 7]
```diff
@@ -1,20 +1,45 @@
...
+dist: trusty
 language: node_js
-sudo: false
+
 node_js:
-- '5.3'
-branches:
-  except:
-  - /^v[0-9\.]+/
+- '6.9.1'
```
Comment (juwara0) [inline .travis.yml]: To work with pr-bumper version 2 this will need to be changed to `6.11.0`

File: .travis.yml [hunk 8]
```diff
@@ -23,22 +48,30 @@ addons:
...
   api_key:
     secure: KP3qIhLdSto+b4GRx0RlJ5ISTPPVVo49doXIRPJxB/QwTG0uepdrWda7DQTdZuYR/YFW08bPXM7UIcT6IQIy7P0Mo/gERyM/hKKfjMuuTjR0DwG9McUW0v37uM4QJ6Uw/v8rg3GilFurrq15KWmiL2Wzcz8vS8G9Y9+4FZVZxMkqzSPsT6IKP3Y8HHKAs5Ueu7jLI7SaaQL+u8go+hG169FmdhLVw6ahOsFaEZCLJgFO1/s1rFL5FWWf0YT0NRDrGB1fiC41QWgnUaHoyvwx8tkdRY2nZoCb1yaw8F/IlzFsiHAUfWXRmihhtgISFY+SqMQlkAyFsSsTZnCKu8ln6aWyr+JEHPGJ2yrI70bbXaQPT9k/gI6oLXOv6KcVHki9t1bT4gnilM5QAgyi8bktRApeVnBcZNy5AH6rb0w669elSJ57PyaXdGV/ChjxY5WaAb7epPKwr/fSL/tMi7URgg2lyArmfRuxqOY3G1m4ff6FFD15pO5dPnkyejpefcwxq/80ssnHhoG5fjP7f0A1wGck9WXXB0+Rvz4I22GDL4g0q2oE+ioRl8a9hK0rmxgJCiTRmVQ9FhunN5lTNcFAzzA/Beg2GrY5vxC5TzDP6LpYwjJoEV9FIkmM+qtXK7zOt4/jWMqNhtwg7wuD9iGl+Qgk4stsKSp0f353vgCLXBo=
   on:
-    branch: master
-    tags: false
-after_deploy:
-- .travis/publish-gh-pages.sh
+    all_branches: true
+    condition: "$EMBER_TRY_SCENARIO = 'default'"
+    node: '6.9.1'
```
Comment (juwara0) [inline .travis.yml]: To work with pr-bumper version 2 this will need to be changed to `6.11.0`

Comment (juwara0) [inline .travis.yml]: Also need to make the changes here as well:
https://github.com/rox163/ember-frost-pick-list/blob/master/.travis/maybe-bump-version.sh#L3
https://github.com/rox163/ember-frost-pick-list/blob/master/.travis/maybe-publish-coverage.sh#L11

File: .travis.yml [hunk 9]
```diff
@@ -1,20 +1,45 @@
...
+- 'stable'
+
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    - google-chrome
+    packages:
+    - google-chrome-stable
+    - g++-4.8
```
Comment (juwara0) [inline .travis.yml]: We want to run this against `firefox: 'latest'` as well:
```
addons:
  apt:
    sources:
    - ubuntu-toolchain-r-test
    - google-chrome
    packages:
    - google-chrome-stable
    - g++-4.8
  firefox: 'latest'
```

Note the indentation is important here.

File: .travis.yml [hunk 10]
```diff
@@ -1,20 +1,45 @@
...
+- .travis/maybe-bump-version.sh
+
 after_success:
-- sed -i -- 's/SF:ember-frost-pick-list\/\(.*\)/SF:addon\/\1.js/' coverage/lcov.info
-  && rm -f coverage/lcov.info--
-- cat coverage/lcov.info | coveralls
+- .travis/maybe-publish-coverage.sh
+- .travis/maybe-publish-gh-pages.sh
+
 addons:
```
Comment (juwara0) [inline .travis.yml]: This section is a duplicate of the one above. It can be removed.

File: package.json [hunk 11]
```diff
@@ -16,42 +16,42 @@
...
     "ember-lodash": "~0.0.6",
-    "ember-frost-test": "^1.0.0",
-    "ember-prop-types": "~3.12.0",
+    "ember-prop-types": "3.14.1",
+    "ember-resolver": "^2.1.0",
+    "ember-spread": "1.2.1",
     "ember-test-utils": "^4.0.0",
     "ember-truth-helpers": "^1.2.0",
-    "ember-try": "^0.2.2",
+    "ember-welcome-page": "^1.0.1",
```
Comment (juwara0) [inline package.json]: This can be removed.

File: package.json [hunk 12]
```diff
@@ -16,42 +16,42 @@
   },
   "repository": "git@github.com:ciena-frost/ember-frost-pick-list.git",
   "engines": {
-    "node": ">= 5.0.0"
+    "node": ">= 6.0.0"
   },
   "author": "",
   "license": "MIT",
   "devDependencies": {
     "broccoli-asset-rev": "^2.5.0",
```
Comment (juwara0) [inline package.json]: On 2.8 this should be `"broccoli-asset-rev": "^2.4.2",`

File: package.json [hunk 13]
```diff
@@ -16,42 +16,42 @@
...
+    "chai-jquery": "^2.0.0",
     "ember-ajax": "0.7.1",
```
Comment (juwara0) [inline package.json]: On 2.8 this should be `"ember-ajax": "^2.0.1",`

File: package.json [hunk 14]
```diff
@@ -16,42 +16,42 @@
...
     "ember-cli": "~2.8.0",
```
Comment (juwara0) [inline package.json]: This should be pinned.

File: package.json [hunk 15]
```diff
@@ -16,42 +16,42 @@
...
     "ember-cli-app-version": "^1.0.0",
+    "ember-cli-chai": "0.3.2",
     "ember-cli-dependency-checker": "^1.2.0",
+    "ember-cli-frost-blueprints": "1.2.1",
     "ember-cli-htmlbars-inline-precompile": "^0.3.1",
     "ember-cli-inject-live-reload": "^1.4.0",
-    "ember-cli-jshint": "^1.0.0",
+    "ember-cli-mocha": "^0.10.0",
     "ember-cli-release": "^0.2.9",
     "ember-cli-sri": "^2.0.0",
```
Comment (juwara0) [inline package.json]: On 2.8 this should be `"ember-cli-sri": "^2.1.0"`

File: package.json [hunk 16]
```diff
@@ -16,42 +16,42 @@
...
     "ember-cli-test-loader": "^1.1.0",
     "ember-cli-uglify": "^1.2.0",
+    "ember-concurrency": "0.7.19",
     "ember-data": "~2.8.0",
+    "ember-disable-proxy-controllers": "^1.0.1",
```
Comment (juwara0) [inline package.json]: This can be removed and `"ember-disable-prototype-extensions": "^1.1.0",` should be added.

File: package.json [hunk 17]
```diff
@@ -61,7 +61,7 @@
   "dependencies": {
     "ember-cli-babel": "^5.1.5",
```
Comment (juwara0) [inline package.json]: On 2.8 this should be `"ember-cli-babel": "^5.1.6"`

File: tests/dummy/config/environment.js [hunk 18]
```diff
@@ -28,7 +28,7 @@ module.exports = function (environment) {
 
   if (environment === 'test') {
     // Testem prefers this...
-    ENV.baseURL = '/'
+    ENV.rootURL = '/'
```
Comment (juwara0) [inline tests/dummy/config/environment.js]: This should be removed.

File: .travis.yml [hunk 19]
```diff
@@ -23,22 +48,30 @@ addons:
...
-before_deploy:
-- pr-bumper bump
+    - CXX=g++-4.8
+    - secure: m3T8OBefKGIJLK1hogS6WvLXDxGSMX539d6Gazgd9D42Q/iF7ni+58fZM8akQOQMAzNssg82Yv29yS+GSudL68+jKHrzLviZ8gc77FwhLagEHOrxFZWPEdjNwNETkocXX03Rda+Bzo9ZaM+RA2jmJfcxxqu9U9rvjHgqNFyqeVLMrtI8BR0vIlyXEnk9lIroBy8Nw8xTFIQkaB3hJlPjbhnYHQSFlRnn8+gtcdu1ZrlGzm8uLHSacLLlcS0fF42tGeN54eGrgpMhhlOe1Kiu+eCIxnq96uoRWwKSfw+8WqidH/0CTcTaYirIXNdxPjUU3AyHeItmOG7TidAbrhfibu8S57OvVy5dL2cxixx8NRZBwmra0HEA0qEdP381K4Re6rJhJNL93B24SWw0UL7X2KmoFBjiPFYoom8VUm9v6rIe0YMtGYp9a6wBzI+Wx5hpdRQrWG9Rj2H6Sbnrfhl2SfMQwCPyHMpIxLfNAwsV4V6smX0w3MqmLHnzWdLj14k0ETOFVjOpcFxVeza/jo7XLW5FdpEgAxSZXDTI8qVMl2ViqeezTo9EKCHBylOEKIT3/usXbWpN5qAQWR7M/qyua31tWDrgsiNFpWncJQaFcRDF30bvULznR+GILEEk9ZwiBIgeGucsAkDEzQiPY62I8tRx7B26OK2X/sBPW09qNIk=
+
+matrix:
+  fast_finish: true
+  allow_failures:
+  - env: EMBER_TRY_SCENARIO=ember-release
```
Comment (juwara0) [inline .travis.yml]: Most likely we will need to allow failures on `stable` as well for the time being.

`- node_js: 'stable'`

File: .travis.yml (hunk 10 above)
Comment (rox163) [inline .travis.yml]: sorry which section does this duplicate? the script block?

Comment (juwara0) [inline .travis.yml]: The section

```
addons:
  apt:
    sources:
    - ubuntu-toolchain-r-test
    - google-chrome
    packages:
    - google-chrome-stable
    - g++-4.8
```

File: tests/.eslintrc.js [hunk 20]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    embertest: true
+  }
+};
```
Comment (juwara0) [inline tests/.eslintrc.js]: Can remove the semicolon at the end.

File: .eslintrc.js [hunk 21]
```diff
@@ -1,3 +1,3 @@
-{
+module.exports = {
   "extends": ["frost-standard"]
 }
```
Comment (juwara0) [inline .eslintrc.js]: For consistency sake, this is how we did the rest of the repos:
```
module.exports = {
  extends: 'frost-standard'
}
```

**Comment by juwara0** [comment]
This `coverage.js` file needs to be added to `tests/dummy/config/`

https://github.com/ciena-frost/ember-frost-core/blob/master/tests/dummy/config/coverage.js

**Comment by juwara0** [comment]
👍

**Pull request merged**